### PR TITLE
Disable database backups if DB is remote

### PR DESCRIPTION
**Pull request by nqb**
# Description
We assume that a remote DB is backup by a tool directly on remote DB

This commit also deprecate the ability to backup a remote DB from a PacketFence server added by #4090

I will an upgrade note to warn about the deprecation even I don't think a lot of users are using this feature.

# Impacts
Backups of remote databases

# Issue
Related to #6396

# Delete branch after merge
YES

# UPGRADE file entries

```asciidoc
=== Remote database backups

Ability to backup a remote database configured in PacketFence has been deprecated. A dedicated tool on the database server itself must be used to backup PacketFence database.
```

### CODE REVIEW

File: addons/backup-and-maintenance.sh [hunk 1]
```diff
@@ -84,6 +84,13 @@ should_backup(){
     # Default choices
     SHOULD_BACKUP=1
     MARIADB_LOCAL_CLUSTER=0
+
+    if [ $MARIADB_REMOTE_CLUSTER -eq 1 ]; then
+        echo "Server not configured to do DB backups: database backup canceled."
+        SHOULD_BACKUP=0
```
Comment (julsemaan) [inline addons/backup-and-maintenance.sh]: No need to set this variable if we're going to exit right below

**Review by julsemaan — COMMENTED** [review]
I think @fdurand should also take a look at this

File: addons/backup-and-maintenance.sh [hunk 2]
```diff
@@ -84,6 +84,13 @@ should_backup(){
     # Default choices
     SHOULD_BACKUP=1
     MARIADB_LOCAL_CLUSTER=0
+
+    if [ $MARIADB_REMOTE_CLUSTER -eq 1 ]; then
+        echo "Server not configured to do DB backups: database backup canceled."
```
Comment (extrafu) [inline addons/backup-and-maintenance.sh]: canceled -> cancelled

Comment (extrafu) [reply addons/backup-and-maintenance.sh]: I would probably just write to be consistent:

"Server is not configured to do database backups: the operation has been cancelled."

Comment (nqb) [reply addons/backup-and-maintenance.sh]: I agree with you @extrafu. I prefer: "Remote database detected: backup should be done on database server itself."

Comment (extrafu) [reply addons/backup-and-maintenance.sh]: Fine to me!

**Comment by nqb** [comment]
I made adjustments requested.

File: addons/backup-and-maintenance.sh [hunk 3]
```diff
@@ -37,10 +37,12 @@ NODE2_HOSTNAME=''
 NODE1_IP=''
 NODE2_IP=''
 
-# Only for MariaDB remote clusters
-MARIADB_REMOTE_CLUSTER=0
-# should be equal to return of hostname command
-MARIADB_BACKUP_SERVER_HOSTNAME=''
+# to detect MariaDB remote DB
+if [ "$DB_HOST" != "localhost" ] && [ "$DB_HOST" != "100.64.0.1" ]; then
```
Comment (fdurand) [inline addons/backup-and-maintenance.sh:41]: if 127.0.0.1 ?

**Review by JeGoi — APPROVED** [review]
Looks good to me with others revue.

File: addons/backup-and-maintenance.sh (hunk 3 above)
Comment (nqb) [reply addons/backup-and-maintenance.sh:41]: Is `127.0.0.1` a value still valid in 12.0.0 ?

I suppose it's when you are using `haproxy-db` in place of `proxysql`

Comment (fdurand) [reply addons/backup-and-maintenance.sh:41]: it´s just in case we forgot to change the value in the conf. Just to prevent no backup just because of that.

Comment (julsemaan) [reply addons/backup-and-maintenance.sh:41]: 127.0.0.1 is not a valid value anymore in pf.conf whether its haproxy-db or proxysql (there is a FATAL checkup in a cluster for that)

The setup will be hosed if that value is left there so I think a non-working backup will be the least of your trouble if 127.0.0.1 is left there

**Comment by extrafu** [comment]
@julsemaan if all is fine can you please merge?

**Pull request merged**